src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.container.jsx

```diff
@@ -219,47 +214,39 @@ export const ContentLayoutContainer = (
         );
     }
 
-    if (loading) {
-        // While loading new results, render current ones loaded during previous render invocation (if any).
-    } else {
-        if (data.jcr && data.jcr.nodeByPath) {
-            // When new results have been loaded, use them for rendering.
-            let nodeTypeName = data.jcr.nodeByPath.primaryNodeType.name;
-            let isSub = nodeTypeName !== 'jnt:page' && nodeTypeName !== 'jnt:contentFolder' && nodeTypeName !== 'jnt:virtualsite';
-            if (!isSub && params.sub && params.sub === true) {
-                setPath(path, {sub: false});
-            } else if (isSub && (!params.sub || params.sub === false)) {
-                setPath(path, {sub: true});
-            }
-        }
+    if (loading || !data.jcr || !data.jcr.nodeByPath) {
+        return <ProgressOverlay/>;
```

Comment:
please do not replace the entire rendered component by a `<ProgressOverlay/>` but use what we do before by applying it as an overlay of the existing component:
https://github.com/Jahia/jcontent/pull/148/files#diff-35f9b839dc1f08d152da39bc2c69a234L248